bus-mapping/src/evm/opcodes.rs

```diff
@@ -225,12 +229,18 @@ pub fn gen_associated_ops(
     opcode_id: &OpcodeId,
     state: &mut CircuitInputStateRef,
     next_steps: &[GethExecStep],
-) -> Result<(), Error> {
+) -> Result<Vec<ExecStep>, Error> {
     let fn_gen_associated_ops = fn_gen_associated_ops(opcode_id);
     fn_gen_associated_ops(state, next_steps)
 }
 
-pub fn gen_begin_tx_ops(state: &mut CircuitInputStateRef) -> Result<(), Error> {
+pub fn gen_begin_tx_ops(state: &mut CircuitInputStateRef) -> Result<ExecStep, Error> {
+    let mut exec_step = ExecStep {
```

Comment:
I added extra functions `new_begin_tx_step` and `new_end_tx_step` in the `CircuitInputStateRef`. Because the logic of creating begin_tx step and end_tx step are rather different, it makes more sense, imo, that just provides a special API for creating these two special steps.